batch/batch/aioclient.py

```diff
@@ -61,30 +158,77 @@ def __init__(self, batch, job_id, attributes=None, parent_ids=None, _status=None
                 return self._status
             j = random.randrange(math.floor(1.1 ** i))
             time.sleep(0.100 * j)
-            # max 4.45s
+            # max 44.5s
             if i < 64:
                 i = i + 1
 
     async def log(self):
-        return await self.batch._client._get(f'/batches/{self.batch_id}/jobs/{self.job_id}/log')
+        return await self._batch._client._get(f'/batches/{self.batch_id}/jobs/{self.job_id}/log')
 
 
 class Batch:
     def __init__(self, client, id, attributes):
         self._client = client
         self.id = id
         self.attributes = attributes
+
+    async def cancel(self):
+        await self._client._patch(f'/batches/{self.id}/cancel')
+
+    async def status(self):
+        return await self._client._get(f'/batches/{self.id}')
+
+    async def wait(self):
+        i = 0
+        while True:
+            status = await self.status()
+            if status['complete']:
+                return status
+            j = random.randrange(math.floor(1.1 ** i))
+            time.sleep(0.100 * j)
+            # max 44.5s
+            if i < 64:
+                i = i + 1
+
+    async def delete(self):
+        await self._client._delete(f'/batches/{self.id}')
+
+
+class BatchBuilder:
+    def __init__(self, client, attributes, callback):
+        doc = {}
+        if attributes:
+            doc['attributes'] = attributes
+        if callback:
+            doc['callback'] = callback
+
+        self._client = client
+        self._doc = doc
         self._job_idx = 0
+        self._job_docs = []
+        self._jobs = []
+        self._submitted = False
+
+    def create_job(self, image, command=None, args=None, env=None, ports=None,
+                   resources=None, tolerations=None, volumes=None, security_context=None,
+                   service_account_name=None, attributes=None, callback=None, parents=None,
+                   input_files=None, output_files=None, always_run=False, pvc_size=None):
+        if self._submitted:
+            raise ValueError("cannot create a job in an already submitted batch")
 
-    async def create_job(self, image, command=None, args=None, env=None, ports=None,
-                         resources=None, tolerations=None, volumes=None, security_context=None,
-                         service_account_name=None, attributes=None, callback=None, parents=None,
-                         input_files=None, output_files=None, always_run=False, pvc_size=None):
         self._job_idx += 1
```

Comment:
_job_idx isn't needed; the array of _job_docs implicitly provides the job idx

Reply:
It's needed to initialize the Job